prov/shm/src/smr_util.c

```diff
@@ -479,24 +479,67 @@ void smr_map_to_endpoint(struct smr_region *region, int64_t id)
 	return;
 }
 
+void smr_unmap_region(const struct fi_provider *prov, struct smr_map *map,
+		     int64_t peer_id)
+{
+	struct smr_region *peer_region;
+	struct smr_peer *peer;
+	struct util_ep *util_ep;
+	struct smr_ep *smr_ep;
+	struct smr_av *av;
+	int ret = 0;
+
+	ofi_spin_lock(&map->lock);
+	peer_region = map->peers[peer_id].region;
+	if (!peer_region)
+		goto unlock;
+
+	peer = &map->peers[peer_id];
+	av = container_of(map, struct smr_av, smr_map);
+	dlist_foreach_container(&av->util_av.ep_list, struct util_ep, util_ep,
+				av_entry) {
+		smr_ep = container_of(util_ep, struct smr_ep, util_ep);
+		smr_unmap_from_endpoint(smr_ep->region, peer_id);
+	}
+
+	if (map->flags & SMR_FLAG_HMEM_ENABLED) {
+		ret = ofi_hmem_host_unregister(peer_region);
+		if (ret)
+			FI_WARN(prov, FI_LOG_EP_CTRL,
+				"unable to unregister shm with iface\n");
+
+		if (peer->pid_fd != -1) {
+			close(peer->pid_fd);
+			peer->pid_fd = -1;
+		}
+	}
+
+	munmap(peer_region, peer_region->total_size);
+	peer_region = NULL;
```

Comment:
This is unnecessary.

Reply:
Just to be clear, `peer_region` is no longer used beyond this line.

Reply:
What he said ^
You can drop the peer_region variable all together since it is stored in peer->region so just grab peer first and only peer->region (which you should be setting to NULL so keep that one).